R/dentist_qc.R

```diff
@@ -39,10 +39,14 @@
 #' }
 #'
 #' @export
-dentist <- function(zScore, LDmat, nSample,
-                    pValueThreshold = 5e-8, propSVD = 0.4, gcControl = FALSE,
-                    nIter = 10, gPvalueThreshold = 0.05, ncpus = 1, seed = 999, 
-                    correct_chen_et_al_bug = TRUE) {
+dentist_impute_single_window <- function(zScore, LDmat, nSample,
+                                         pValueThreshold = 5e-8, propSVD = 0.4, gcControl = FALSE,
+                                         nIter = 10, gPvalueThreshold = 0.05, ncpus = 1, seed = 999, correct_chen_et_al_bug = TRUE) {
+  # Check that number of variants cannot be below 2000
+  if (length(zScore) < 2000) {
+    warning("The number of variants is below 2000. The function will not be executed.")
+    return(NULL) # Return NULL to indicate no results
```

Comment:
Let's not do this. Just a warning is enough. Let's say that the number is < 2000 the algorithm may not work as expected, as suggested by the DENTIST authors.

Reply:
removed NULL and fixed the message.